lib/ui/common/AlertWidget.dart

```diff
@@ -0,0 +1,242 @@
+    Color actionBackgroundColor,
+    Color destructiveActionBackgroundColor,
+    double actionLineSpace,
+    TextStyle actionTextStyle,
+  });
+}
+
+const AlertWidgetStyle _defaultStyle = const _DefaultStyle();
+
+class AlertWidget extends StatelessWidget {
+  final AlertWidgetViewModel _viewModel;
+  final AlertWidgetStyle _style;
+
+  const AlertWidget(
+      {Key key,
+      AlertWidgetViewModel viewModel,
+      AlertWidgetStyle style = _defaultStyle})
+      : this._viewModel = viewModel,
+        this._style = style,
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Material(
+      type: MaterialType.transparency,
+      child: Center(
+        child: Container(
+          margin: EdgeInsets.symmetric(horizontal: 24),
+          decoration: BoxDecoration(
+            borderRadius: _style.cornerRadius,
+            color: _style.backgroundColor,
+          ),
+          child: Padding(
+            padding: _style.alertInnerPadding,
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              mainAxisSize: MainAxisSize.min,
+              children: <Widget>[
+                Text(
+                  _viewModel.titleText,
+                  style: _style.titleTextStyle,
+                ),
+                SizedBox(
+                  height: _style.titleLineSpace,
+                ),
+                Column(
+                  mainAxisSize: MainAxisSize.min,
+                  children: <Widget>[
+                    Text(
+                      _viewModel.detailText,
+                      style: _style.detailTextStyle,
+                    ),
+                    SizedBox(
+                      height: _style.detailLineSpace,
+                    ),
+                    Row(
+                      children: _buildAction(context),
+                    )
+                  ],
+                )
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  confirmDialog(BuildContext context) {
+    return showDialog<bool>(
+        context: context,
+        barrierDismissible: false,
+        builder: (_) => AlertWidget());
+  }
+
+  List<Widget> _buildAction(BuildContext context) {
+    List<Widget> listBuilder = [
+      Expanded(
+        child: RoundedButton(
+          viewModel: RoundedButtonViewModel(
+              title: _viewModel.cancelButtonTittle,
+              onTap: () => Navigator.pop(context)),
+          style: RoundedButtonStyle.actionSheetLargeRoundedButton().copyWith(
+              height: _style.actionHeight,
+              width: _style.actionWidth,
+              borderRadius: _style.actionCornerRadius),
+        ),
+      )
+    ];
+    listBuilder.add(SizedBox(
+      width: _style.actionLineSpace,
+    ));
+    listBuilder.add(Expanded(
+      child: RoundedButton(
+        viewModel: RoundedButtonViewModel(
+            title: _viewModel.confirmButtonTittle, onTap: () {}),
```

Comment:
pass the action to `onTap:`

Reply:
also always put in a mock action. even if its just a console print. Inject it via the viewmodel so that when its used we don't have to edit the widget to remove any test code.

Reply:
Yeah I added that today in early morning, i just commit that ;)